include/tut/tut_assert.hpp

```diff
@@ -149,10 +168,20 @@ void ensure_equals(const M& msg, const double& actual, const double& expected, c
     }
 }
 
+#if __cplusplus >= 201103L
+template<typename M, typename LHS, typename RHS>
+typename std::enable_if<std::is_floating_point<LHS>::value || std::is_floating_point<RHS>::value, void>::type
+    ensure_equals(const M& msg, const LHS& actual, const RHS& expected)
+{
+	auto epsilon = sizeof(LHS) < sizeof(RHS) ? std::numeric_limits<LHS>::epsilon() : std::numeric_limits<RHS>::epsilon();
```

Comment:
The goal is here to get the biggest epsilon if `LHS` and `RHS` are not of the same type